src/main/java/com/example/bankProjetc/controller/TransationController.java

```diff
@@ -0,0 +1,56 @@
+package com.example.bankProjetc.controller;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.example.bankProjetc.Views.Views;
+import com.example.bankProjetc.model.Transation;
+import com.example.bankProjetc.services.TransationService;
+import com.fasterxml.jackson.annotation.JsonView;
+
+@RestController
+public class TransationController {
+	@Autowired
+	TransationService transatioService;
+
+	@PostMapping("/transaction")
+	public ResponseEntity<Transation> createTransation(@RequestBody @JsonView(value = Views.TransactionView.post.class) Transation transation) {
+		return new ResponseEntity<Transation>(transatioService.createTranation(transation), HttpStatus.OK);
+	}
+
+	@GetMapping("/transations")
+	public List<Transation> getAllTransations() {
+		return transatioService.findAllTransations();
+	}
+
+	@GetMapping("/transations/{id}")
+	public ResponseEntity<Transation> getTransationById(@PathVariable("id") long TransationId) {
+		return new ResponseEntity<Transation>(transatioService.findTransationByTransationId(TransationId),
+				HttpStatus.OK);
+
+	}
+
+	@PutMapping("/transation/{id}")
```

Comment:
add json view on get and put apis as well, use different views for get and put ( to be created  for each of Account/ transaction and customer)